File: gameplay/js/model/Hex.js
/**
Hex Class 
<pre>
</pre>
@class Hex 
@constructor
@param {boolean} isLand
@param {HexLocation} location
@param {Edge[]} edges
@param {Vertex[]} vertexes
*/
function Hex(isLand, location) {
	this.isLand = isLand;
	this.location = location;
}

/**
GetEdges method
<pre>
</pre>
@method getEdges
@return {Edge[]} edges
*/
Hex.prototype.getEdges = function() {
    return this.edges;
}

/**
GetLocation method
<pre>
</pre>
@method getLocation
@return {HexLocation} location
*/
Hex.prototype.getLocation = function() {
    return this.location;
}

/**
GetVertexes method
<pre>
</pre>
@method getVertexes
@return {Vertex[]} vertexes
*/
Hex.prototype.getVertexes = function() {
    return this.vertexes;
}

/**
IsLand method
<pre>
</pre>
@method isLand
@return {boolean} isLand
*/
Hex.prototype.isLand = function() {
    return this.isLand;
}

/**
SetEdge method
<pre>
</pre>
@method setEdge
@param {Edge} edge
*/
Hex.prototype.setEdge = function(edge) {
    this.edges.push(edge);
}

/**
SetVertex method
<pre>
</pre>
@method setVertex
@param {Vertex} vertex
*/
Hex.prototype.setVertex = function(vertex) {
    this.vertexes.push(vertex);
}

File: server/src/catan/model/Hex.java
package catan.model;

import java.util.Arrays;

public class Hex {

	private Edge[] edges;
	private Vertex[] vertexes;
	private HexLocation location;
	private String landtype;
	private boolean isLand;
	
	public Hex(int x, int y) {
		
		edges = new Edge[6];
		vertexes = new Vertex[6];
		for (int i = 0; i < 6; i++) {
			
			edges[i] = new Edge();
			vertexes[i] = new Vertex();
		}
		
//		location = new HexLocation(x, y);
	}

	public Edge[] getEdges() {
		return edges;
	}

	public void setEdges(Edge[] edges) {
		this.edges = edges;
	}

	public Vertex[] getVertexes() {
		return vertexes;
	}

	public void setVertexes(Vertex[] vertexes) {
		this.vertexes = vertexes;
	}

	public HexLocation getLocation() {
		return location;
	}

	public void setLocation(HexLocation location) {
		this.location = location;
	}
	
	public void setType(String type) {
		
		this.landtype = type;
		isLand = true;
		if (type.equals("water"))
			isLand = false;	
	}
	
	public String getType() {
		
		return landtype;
	}
	
	public boolean isLand() {
		
		return isLand;
	}
	
	@Override
	public String toString() {
		return "Hex [edges=" + Arrays.toString(edges) + ", vertexes="
				+ Arrays.toString(vertexes) + ", location=" + location
				+ ", landtype=" + landtype + ", isLand=" + isLand + "]";
	}
}
